documentation/general/dotnet-run-file.md

```diff
@@ -203,10 +226,12 @@ Because these directives are limited by the C# language to only appear before th
 dotnet CLI can look for them via a regex or Roslyn lexer without any knowledge of defined conditional symbols
 and can do that efficiently by stopping the search when it sees the first "C# token".
 
-We do not limit these directives to appear only in entry point files.
-Indeed, it might be beneficial to let a non-entry-point file like `Util.cs` be self-contained and have all the `#:package`s it needs specified in it,
-which also makes it possible to share it independently or symlink it to multiple script folders.
-This is also similar to `global using`s which users usually put into a single file but don't have to.
+For a given `dotnet run file.cs`, we include directives from the current entry point file (`file.cs`) and all other non-entry-point files.
+The order in which other files are processed is currently unspecified (can change across SDK versions) but deterministic (stable in a given SDK version).
```

Comment:
Why the lack of specification? Because doesn't the order of SDK inclusion impact the MSBuild property evaluation and such?

Reply:
> doesn't the order of SDK inclusion impact the MSBuild property evaluation and such?

Yes it does, but I hope users won't rely on that in their simple file-based programs. We can specify this, but once we do, any changes will be breaking, so that's why I chose to not specify at first.

Reply:
Right, but only if someone chooses to rely on undefined behavior in the first place.

Anyway, I can specify some ordering here, but maybe it would be better to wait a bit and see what makes most sense when we see real-world usage. For example, I'm currently unsure whether it makes sense for entry-point directives to go before or after non-entry-point directives? Should other files be just alphabetically ordered, or should their directory nesting also play a role in the order?